pyctbgui/utils/numpyWriter/npy_writer.py

```diff
@@ -0,0 +1,183 @@
+"""
+Wrapper to be able to append frames to a numpy file
+
+numpy header v1
+
+- 6bytes                         \x93NUMPY
+- 1 byte major version number    \x01
+- 1 byte minor version number    \x00
+- 2 bytes (unsigned short) HEADER_LEN length of header to follow
+- Header as an ASCII dict terminated by \n padded with space \x20 to make sure
+we get len(magic string) + 2 + len(length) + HEADER_LEN divisible with 64
+Allocate enough space to allow for the data to grow
+"""
+
+import ast
+import logging
+import os
+import zipfile
+from pathlib import Path
+
+import numpy as np
+
+
+class NumpyFileManager:
+    """
+    class used to read and write into .npy files that can't be loaded completely into memory
+    """
+    magic_str = np.lib.format.magic(1, 0)
+    headerLength = np.uint16(128)
+    FSEEK_FILE_END = 2
+    BUFFER_MAX = 500
+
+    def __init__(
+        self,
+        file: str | Path | zipfile.ZipExtFile,
+        frameShape: tuple = None,
+        dtype=None,
+        resetFile: bool = False,
```

Comment:
I would prefer to use a mode like open (https://docs.python.org/3/library/functions.html#open) which would be more intuitive for many people.
```
'r' = read default option
'w'=write, by default truncates existing files
'x'=exclusive creation for write, fails if file exists
```
```python
f = NumpyFileManager('somefile.npy') #opens
f = NumpyFileManager('somefile.npy', 'w') #for writing etc

```